packages/db/src/connection.ts

```diff
@@ -0,0 +1,129 @@
+import Knex from 'knex';
+import { getSecretConnectionConfig, getEnvConnectionConfig } from './config';
+
+import { envConectionConfigObject, knexSecretConnectionConfigObject } from './types';
+
+export interface knexConnectionConfigObject {
+  host: string,
+  username: string,
+  password: string,
+  database?: string,
+}
+
+interface knexConfigObject {
+  client: string,
+  connection: knexConnectionConfigObject,
+  acquireConnectionTimeout: number,
+  asyncStackTraces: boolean,
+  debug: boolean,
+  migrations?: {
+    directory: string
+  }
+}
+
+interface knexEnvironmentObject extends NodeJS.ProcessEnv {
+  KNEX_DEBUG?: string
+  KNEX_ASYNC_STACK_TRACES?: string,
+}
+
+/**
+* @summary Given a knexEnvironmentObject and a Knex.PgConnectionConfig, returns
+* a Knex instance with those confiugrations applied
+*
+* @param {Knex.PgConnectionConfig} connectionConfig - Knex connection configuration
+* @param {knexEnvironmentObject} env - environment object with knex configuration keys
+* @returns {Knex} - configured Knex instance
+*/
+const getConfiguredKnex = (
```

Comment:
This function feels a little weird, I'm trying to put my finger on it. It's taking a connection config, environment variables, and default values and building a knex config. It then returns a Knex connection object. I guess my concern is two things. It feels like we're passing in parameters that are at the wrong level of abstraction, and we're doing two things (creating a config and creating a connection).

I think that I'd feel a little better if, rather than passing in the `env` object, this function took named params of `{ connectionConfig, debug = false, timeout = 60000, etc }`. It might also be a little more clear if this function just built the knex config.

Reply:
Tried it out, and think I like that better  3cc1f6f1f - the obvious downside to moving the env up is you duplicate the param definitions, and really there's no good reason to have multiple interfaces to this, so: a74e7bc28